### PR TITLE
Add API linking integration test

### PR DESCRIPTION
**Pull request by desimmons**
## Checklist before requesting a review
- [ ] I have performed a self-review of my code

### CODE REVIEW

File: lib/marketplace/marketplace_home_screen.dart [hunk 1]
```diff
@@ -1023,6 +1024,8 @@ class _AssetScreenState extends State<MarketplaceHomeScreen>
       String entryPointUrlRoot,
       String exceptionString,
       void Function(String) onNFTOfferPoll) {
+    Logger logger = Logger();
+    logger.d("EIJNIWJDNWJNWCO");
```
Comment (desimmons) [inline lib/marketplace/marketplace_home_screen.dart:1028]: Please remove

File: lib/marketplace/marketplace_dialogs.dart [hunk 2]
```diff
@@ -889,7 +889,7 @@ class _DropzoneDeployWidgetState extends State<DropzoneDeployWidget> {
           // The callback is passed right through to
           // `HtmlElementView` (here)[https://github.com/deakjahn/flutter_dropzone/blob/master/flutter_dropzone_web/lib/flutter_dropzone_plugin.dart#L117]
           // which suggests that the platform view does not get created during the test.
-          if (controller != null) {
+          if (false) {
```
Comment (desimmons) [inline lib/marketplace/marketplace_dialogs.dart:892]: Undo this

File: lib/marketplace/marketplace_dialogs.dart [hunk 3]
```diff
@@ -221,7 +221,7 @@ class _DropzoneRunWidgetState extends State<DropzoneRunWidget> {
...
```
Comment (desimmons) [inline lib/marketplace/marketplace_dialogs.dart:224]: Undo this

File: integration_test/hit_linked_api.py [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+def main():
+    name = "integration-test-api"
+    documents = find_document_by_field("name", name)
+    documents[0].id
+    url = f"https://dhali-staging-run-3mmgxhct.uc.gateway.dev/{documents[0].id}/xls20-nfts/all/issuers"
+
+    print(url)
+
+    PAYMENT_CLAIM = '{"account": "rBqZz7QjHLizCHGrfAAh5cZtvG7NS3DvBV", "destination_account": "rstbSTpPcyxMsiXwkBxS9tFTrg2JsDNxWk", "authorized_to_claim": "10000000", "signature": "47D6A8BFCB754BC859EF897F3DE0123F35B58F941E664B74E923766FE0E1AD2E3F655F1AB16954096442E74878EFF1C875D34EFB132D513D611F9615DDCBD70B", "channel_id": "742A4A95984FB1325FC882DFB88254102A333A6156FFFAB514174A5F8FB06B25"}'
```
Comment (robbie-daniels) [inline integration_test/hit_linked_api.py:24]: Will this payment claim eventually get used up?

File: integration_test/linking_integration_test.dart [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+  await tester.pumpAndSettle();
+  await tester.tap(find.text("My APIs"));
+  expect(find.text(theInputAssetName), findsWidgets);
+}
+
+void main() {
+  XRPLWallet wallet = XRPLWallet(
+      "romance acoustic essay simple bless fluid symptom code very tip text excess",
+      getFirestore: () => FirebaseFirestore.instance,
+      testMode: true);
```
Comment (robbie-daniels) [inline integration_test/linking_integration_test.dart:116]: Do we want to make sure this is always funded?

File: lib/main.dart [hunk 6]
```diff
@@ -43,13 +43,14 @@ void main() async {
 }
 
 class MyApp extends StatefulWidget {
-  const MyApp({super.key, required this.getRequest});
+  const MyApp({super.key, required this.getRequest, this.wallet});
```
Comment (robbie-daniels) [inline lib/main.dart:46]: How come we're injecting the wallet now?

File: lib/marketplace/marketplace_dialogs.dart [hunk 7]
```diff
@@ -2302,6 +2304,12 @@ Widget getDialogTemplate(
     required BuildContext context,
     int? step,
     int? steps}) {
+  Key closeDialogKey;
+  if (step != null && steps != null) {
+    closeDialogKey = Key("CloseDialog-$step-$steps");
+  } else {
+    closeDialogKey = const Key("CloseDialog");
+  }
```
Comment (robbie-daniels) [inline lib/marketplace/marketplace_dialogs.dart:2312]: A comment here would be helpful!

**Review by robbie-daniels — COMMENTED** [review]
* Any reason why `hit_linked_api.py` isn't done as a PyTest?  Same for `linking_integration_test.dart` with `flutter_test`.  Is `flutter_test` incompatible with `integration_driver`?

File: lib/marketplace/marketplace_dialogs.dart (hunk 7 above)
Comment (robbie-daniels) [reply lib/marketplace/marketplace_dialogs.dart:2312]: The comment would explain: Why it's here, and necessary for the integration tests

**Comment by desimmons** [comment]
@robbie-daniels  for the `flutter test` comment you made above, I'm just following the [docs](https://docs.flutter.dev/testing/integration-tests?gclid=CjwKCAiAsIGrBhAAEiwAEzMlC0PZuotESzPOyhQgxcXWC1A3ji_qZQBi9ATy_UXIA0F_GJtulSs77RoCU0UQAvD_BwE&gclsrc=aw.ds) for Flutter integration tests, which say to use `flutter drive`

File: integration_test/hit_linked_api.py (hunk 4 above)
Comment (desimmons) [reply integration_test/hit_linked_api.py:24]: Added automatic claim generation

File: lib/main.dart (hunk 6 above)
Comment (desimmons) [reply lib/main.dart]: Removed this now

File: .github/workflows/linking-integration-test.yml [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+      run: |
+        pip install google-cloud-firestore firebase-admin dhali-py
+
+    - name: Hit created API test
+      run: |
+        PYTHONPATH=integration_test/ pytest integration_test/test_hit_linked_api.py
+
+    - name: Cleanup back-end
+      run: |
+        PYTHONPATH=integration_test/ pytest integration_test/test_validate_and_clean_firestore.py
```
Comment (robbie-daniels) [inline .github/workflows/linking-integration-test.yml:59]: Nit: newline

**Review by robbie-daniels — APPROVED** [review]

**Pull request merged**